clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -274,11 +274,12 @@ public class ProducerConfig extends AbstractConfig {
 
     /** <code>retries</code> */
     public static final String RETRIES_CONFIG = CommonClientConfigs.RETRIES_CONFIG;
-    private static final String RETRIES_DOC = "Setting a value greater than zero will cause the client to resend any record whose send fails with a potentially transient error."
-            + " Note that this retry is no different than if the client resent the record upon receiving the error."
-            + " Produce requests will be failed before the number of retries has been exhausted if the timeout configured by"
-            + " <code>" + DELIVERY_TIMEOUT_MS_CONFIG + "</code> expires first before successful acknowledgement. Users should generally"
-            + " prefer to leave this config unset and instead use <code>" + DELIVERY_TIMEOUT_MS_CONFIG + "</code> to control"
+    private static final String RETRIES_DOC = "Number of times to retry a request that fails with a transient error."
+            + " Setting a value greater than zero will cause the client to resend any record whose send fails with a potentially transient error. "
+            + " Requests will be retried this many times until they succeed, fail with a non-transient error, or the <code> DELIVERY_TIMEOUT_MS_CONFIG </code> expires."
```

Comment:
```suggestion
            + " Requests will be retried this many times until they succeed, fail with a non-transient error, or the <code>DELIVERY_TIMEOUT_MS_CONFIG</code> expires."
```

Reply:
Well, should this actually be `<code>" + DELIVERY_TIMEOUT_MS_CONFIG + "</code>` ?